pkg/controllers/mcs/mcs_controller.go

```diff
@@ -0,0 +1,398 @@
+			Name:       names.GenerateWorkName(svc.Kind, svc.Name, clusterName+"/"+svc.Namespace),
+			Namespace:  names.GenerateExecutionSpaceName(clusterName),
+			Finalizers: []string{util.ExecutionControllerFinalizer},
+			Labels: map[string]string{
+				networkingv1alpha1.MultiClusterServicePermanentIDLabel: mcsID,
+				util.ManagedByKarmadaLabel:                             util.ManagedByKarmadaLabelValue,
+			},
+		}
+
+		if err = helper.CreateOrUpdateWork(c, workMeta, &unstructured.Unstructured{Object: svcObj}); err != nil {
+			klog.Errorf("Failed to create or update resource(%v/%v) in the given member cluster %s, err is %v",
+				workMeta.GetNamespace(), workMeta.GetName(), clusterName, err)
+			c.EventRecorder.Event(mcs, corev1.EventTypeWarning, events.EventReasonSyncServiceWorkFailed, fmt.Sprintf(
+				"Failed to create or update resource(%v/%v) in member cluster(%s): %v",
+				workMeta.GetNamespace(), workMeta.GetName(), clusterName, err))
+			errs = append(errs, err)
+		}
+	}
+	if len(errs) != 0 {
+		return syncClusters, errors.NewAggregate(errs)
+	}
+
+	return syncClusters, nil
+}
+
+func (c *MCSController) updateMCSStatus(
+	ctx context.Context,
+	mcs *networkingv1alpha1.MultiClusterService,
+) error {
+	currentServiceAppliedCondition := meta.FindStatusCondition(mcs.Status.Conditions, MCSServiceAppliedConditionType)
+	if currentServiceAppliedCondition == nil {
+		mcs.Status.Conditions = append(mcs.Status.Conditions, metav1.Condition{
+			Message:            "Service is propagated to target clusters.",
+			Reason:             "ServiceAppliedSuccess",
+			Status:             metav1.ConditionTrue,
+			Type:               MCSServiceAppliedConditionType,
+			LastTransitionTime: metav1.Now(),
+		})
+	}
+
+	return retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
+		updateErr := c.Status().Update(ctx, mcs)
+		if updateErr == nil {
+			return nil
+		}
+
+		updated := &networkingv1alpha1.MultiClusterService{}
+		if err = c.Get(ctx, client.ObjectKey{Namespace: mcs.Namespace, Name: mcs.Name}, updated); err == nil {
+			updated.Status.Conditions = mcs.Status.Conditions
+			mcs = updated
+		} else {
+			klog.Errorf("Failed to get updated MultiClusterService(%s/%s): %v", mcs.Namespace, mcs.Name, err)
+		}
+		return updateErr
+	})
+}
+
+func (c *MCSController) getMultiClusterServiceID(mcs *networkingv1alpha1.MultiClusterService) (string, error) {
+	id := util.GetLabelValue(mcs.GetLabels(), networkingv1alpha1.MultiClusterServicePermanentIDLabel)
+	if id == "" {
+		id = uuid.New().String()
+		mcs.Labels = util.DedupeAndMergeLabels(mcs.Labels, map[string]string{networkingv1alpha1.MultiClusterServicePermanentIDLabel: id})
+		if err := c.Client.Update(context.TODO(), mcs); err != nil {
+			return id, err
+		}
+	}
+
+	return id, nil
+}
+
+// MCSContainCrossClusterType checks weather the MultiClusterService contains CrossCluster type.
+func MCSContainCrossClusterType(mcs *networkingv1alpha1.MultiClusterService) bool {
```

Comment:
Let's put this in file https://github.com/karmada-io/karmada/blob/02b9ff4e11f6bbf67655335fd3c4d1b45f223b25/pkg/util/helper/mcs.go#L103

Reply:
ok